tests/assets/test_digital_assets.py

```diff
@@ -0,0 +1,163 @@
+    tx_transfer_signed = tx_transfer.sign([user_sk])
+
+    assert tx_transfer_signed.validate(b) == tx_transfer_signed
+    assert tx_transfer_signed.asset.data_id == tx_create.asset.data_id
+
+
+def test_validate_bad_asset_creation(b, user_vk):
+    from bigchaindb_common.exceptions import AmountError
+    from bigchaindb.models import Transaction
+
+    # `divisible` needs to be a boolean
+    tx = Transaction.create([b.me], [user_vk])
+    tx.asset.divisible = 1
+    tx_signed = tx.sign([b.me_private])
+    with pytest.raises(TypeError):
+        tx_signed.validate(b)
+
+    # `refillable` needs to be a boolean
+    tx = Transaction.create([b.me], [user_vk])
+    tx.asset.refillable = 1
+    tx_signed = tx.sign([b.me_private])
+    with pytest.raises(TypeError):
+        b.validate_transaction(tx_signed)
+
+    # `updatable` needs to be a boolean
+    tx = Transaction.create([b.me], [user_vk])
+    tx.asset.updatable = 1
+    tx_signed = tx.sign([b.me_private])
+    with pytest.raises(TypeError):
+        b.validate_transaction(tx_signed)
+
+    # `data` needs to be a dictionary
+    tx = Transaction.create([b.me], [user_vk])
+    tx.asset.data = 'a'
+    tx_signed = tx.sign([b.me_private])
+    with pytest.raises(TypeError):
+        b.validate_transaction(tx_signed)
+
+    # TODO: Check where to test for the amount
+    """
+    tx = b.create_transaction(b.me, user_vk, None, 'CREATE')
+    tx['transaction']['conditions'][0]['amount'] = 'a'
+    tx['id'] = get_hash_data(tx['transaction'])
+    tx_signed = b.sign_transaction(tx, b.me_private)
+    with pytest.raises(TypeError):
+        b.validate_transaction(tx_signed)
+
+    tx = b.create_transaction(b.me, user_vk, None, 'CREATE')
+    tx['transaction']['conditions'][0]['amount'] = 2
+    tx['transaction']['asset'].update({'divisible': False})
+    tx['id'] = get_hash_data(tx['transaction'])
+    tx_signed = b.sign_transaction(tx, b.me_private)
+    with pytest.raises(AmountError):
+        b.validate_transaction(tx_signed)
+
+    tx = b.create_transaction(b.me, user_vk, None, 'CREATE')
+    tx['transaction']['conditions'][0]['amount'] = 0
+    tx['id'] = get_hash_data(tx['transaction'])
+    tx_signed = b.sign_transaction(tx, b.me_private)
+    with pytest.raises(AmountError):
+        b.validate_transaction(tx_signed)
+    """
+
+@pytest.mark.usefixtures('inputs')
+def test_validate_transfer_asset_id_mismatch(b, user_vk, user_sk):
+    from bigchaindb_common.exceptions import AssetIdMismatch
+    from bigchaindb.models import Transaction
+
+    tx_create = b.get_owned_ids(user_vk).pop()
+    tx_create = b.get_transaction(tx_create.txid)
+    tx_transfer = Transaction.transfer(tx_create.to_inputs(), [user_vk], tx_create.asset)
+    tx_transfer.asset.data_id = 'aaa'
+    tx_transfer_signed = tx_transfer.sign([user_sk])
+    with pytest.raises(AssetIdMismatch):
+        tx_transfer_signed.validate(b)
+
+
+@pytest.mark.usefixtures('inputs')
```

Comment:
We don't necessarily need inputs here.
We could just create the transactions without writing them to the database as `get_asset_id` doesn't need the db.